tests/cli/test_ci.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+# Note: execute this file from the project root directory
+
+set -e
+
+. /usr/share/beakerlib/beakerlib.sh
+. $(dirname $0)/lib/lib.sh
+
+CLI="${CLI:-./src/bin/composer-cli}"
+
+dnf -y install \
+  anaconda-tui \
+  libgit2-glib \
+  libselinux-python3 \
+  make \
+  pykickstart \
+  python3-coverage \
+  python3-coveralls \
+  python3-flask \
+  python3-gevent \
+  python3-magic \
+  python3-mako \
+  python3-mock \
+  python3-nose \
+  python3-pocketlint \
+  python3-pycdlib \
+  python3-pylint \
+  python3-pyparted \
+  python3-toml \
+  python3-semantic_version \
+  python3-sphinx \
+  python3-sphinx-argparse \
+  python3-sphinx_rtd_theme \
+  python3-rpmfluff \
+  python3-librepo \
+  python3-pyvmomi \
+  beakerlib \
+  sudo \
+  rsync \
+  e2fsprogs \
+  xz-lzma-compat \
+  pbzip2 \
+  squashfs-tools \
+  qemu-img \
+  which 
+
+mkdir /lorax
+git clone https://github.com/weldr/lorax /lorax/
```

Comment:
this isn't paying attention to the current branch or PR commit that is being tested.

Reply:
If we continue going the same way as now there should be some way to mount the current git checkout into the VM and just call make ci from there. The CI slave system (be it Travis or Cockpit) already has a local git checkout before even executing any of this code.

Reply:
In my split the tests branch I ended up adding a bit of code to the class that copies the tests into the VM using `self.machine.upload(["../tests"], "/")` that could be used to copy the whole checkout if needed.

Reply:
Running against HEAD instead of the current checkout is not good. Even when the failures reported in the logs are fixed this job will still report FAIL b/c it won't pick up the changes.

@jrusz `self.machine.upload(["../"], "/lorax")` should be able to copy the whole git checkout from the CI host to the VM. Then inside the VM you just `cd /lorax && make ci`. Give it a try.

Reply:
I solved this in commit 0a486514375f79eb79d0355946ab78bcb53c9495 .. There are two reasons for why I didn't use exactly self.machine.upload(["../"], "/lorax") .. First is that when I tried it I got an error saying that "scp: error: unexpected filename: ..". What could work is self.machine.upload(["../../lorax"], "/lorax") but I can't really do that on my machine because I have the bots repo there too. So I decided to just upload the relevant files I hope I got all of them.

Reply:
> but I can't really do that on my machine because I have the bots repo there too.

I don't understand what the problem is and how does having bots repo on your development machine prevents you from doing something in CI ?

Reply:
It doesn't prevent me from doing something in CI but it prevents me from testing it on my machine first. What I mean is that I can't upload my whole local lorax repo because I have the bots repo inside for running the vm locally. Meaning that when I would want to run some test locally and then upload the whole lorax folder inside, it would also try to copy the whole bots folder along with the images. AFAIK I need to have it there if I want to run it locally first.

Reply:
You can always upload the entire dir inside the VM and remove the bots/ directory from the VM if it causes failures (may probably affect only pylint). If it doesn't cause any problems when running inside the VM I see no reason to worry about it.